source/slang/slang-constructor-utility.cpp

```diff
@@ -0,0 +1,264 @@
+            invoke->functionExpr = visitor->ConstructDeclRefExpr(member, nullptr, defaultCtor->getName(), defaultCtor->loc, nullptr);
+            invoke->type = varDeclType.type;
+            return invoke;
+        }
+        else
+        {
+            return createDefaultConstructExprForType(visitor->getASTBuilder(), QualType(varDeclType.type), {});
+        }
+    }
+
+    FuncDecl* findZeroInitListFunc(StructDecl* structDecl)
+    {
+        for (auto funcDecl : structDecl->getMembersOfType<FuncDecl>())
+        {
+            if (!funcDecl->findModifier<ZeroInitModifier>())
+                continue;
+            return funcDecl;
+        }
+        return nullptr;
+    }
+
+    Expr* _constructZeroInitListFuncMakeDefaultCtor(SemanticsVisitor* visitor, StructDecl* structDecl, Type* structDeclType, ConstructorDecl* defaultCtor)
+    {
+            auto* invoke = visitor->getASTBuilder()->create<InvokeExpr>();
+            auto member = visitor->getASTBuilder()->getMemberDeclRef(structDecl->getDefaultDeclRef(), defaultCtor);
+            invoke->functionExpr = visitor->ConstructDeclRefExpr(member, nullptr, defaultCtor->getName(), defaultCtor->loc, nullptr);
+            invoke->type = structDeclType;
+            return invoke;
+    }
+    Expr* constructZeroInitListFunc(SemanticsVisitor* visitor, StructDecl* structDecl, Type* structDeclType, ConstructZeroInitListOptions options)
+    {
+        SLANG_ASSERT(structDecl);
+
+        // 1. Prefer non-synth default-ctor
+        //  * Skip this option if `ConstructZeroInitListOptions::PreferZeroInitFunc` is true
+        //  * Skip this option if `ConstructZeroInitListOptions::CheckToAvoidRecursion` detects recursion
+        //      * Only user-defined ctor will try and have recursion of `{}`
+        // 2. Prefer $ZeroInit
+        // 3. Prefer any default-ctor
+        // 4. Use `DefaultConstructExpr`
+
+        // 1.
+        auto defaultCtor = _getDefaultCtor(structDecl);
+        if(defaultCtor
+            && !defaultCtor->containsOption(ConstructorTags::Synthesized)
+            && !((UInt)options & (UInt)ConstructZeroInitListOptions::PreferZeroInitFunc))
+        {
+            bool canCreateCtor = true;
+            if(((UInt)options & (UInt)ConstructZeroInitListOptions::CheckToAvoidRecursion))
+            {
+                auto callingScope = visitor->getOuterScope();
+                while (callingScope)
+                {
+                    if (callingScope->containerDecl == defaultCtor)
+                    {
+                        canCreateCtor = false;
+                        break;
+                    }
+                    callingScope = callingScope->parent;
+                }
+            }
+            if(canCreateCtor)
+                return _constructZeroInitListFuncMakeDefaultCtor(visitor, structDecl, structDeclType, defaultCtor);
+        }
+
+        // 2.
+        if (auto zeroInitListFunc = findZeroInitListFunc(structDecl))
```

Comment:
al `zeroInitListFunc` should be named `defaultInitFunc`.

Reply:
addressed in: https://github.com/shader-slang/slang/pull/5107/commits/d966b3fc88a24a4f45780b651bb918a2c216889c